dcicutils/ff_utils.py

```diff
@@ -457,65 +457,150 @@ def faceted_search(key=None, ff_env=None, item_type=None, **kwargs):
     return search_metadata(search, ff_env=ff_env, key=key)
 
 
-def get_associated_qc_metrics(uuid, key=None, ff_env=None,
-                              exclude_raw_files=True,
-                              exclude_supplementary_files=True):
+def fetch_files_qc_metrics(data, associated_files=['processed_files'], ignored_fields=True, key=None, ff_env=None):
     """
-    Given a uuid of an experiment set, return a dictionary of uuid : item
-    mappings of quality metric items.
+    Utility function to grap all the qc metrics from associated types of file such as:
+    'proccessed_files', 'other_processed_files', 'files'
+    inputs:
+        data: the metadata of a ExperimentSet or Experiment
+        associated_files: a list of the types of the files fields the qc metrics will be extracted from:
+            examples are = ['files', 'processed_files', 'other_processed_files']
+    Args:
+        ignored_fields: flag to ignore 4DN custom fields from the qc metric object
+
+    Returns:
+        a dictionary of dictionaries containing the qc_metric information
+    """
+    qc_metrics = {}
+
+    if ignored_fields:
+        ignored_qc_fields = ['contributing_labs', 'schema_version', 'external_references', '@context', 'aliases',
+                             'project_release', 'award', 'principals_allowed', 'validation-errors',
+                             'last_modified', 'slope', '@id', 'aggregated-items', 'status', 'public_release',
+                             'actions', 'submitted_by', 'convergence', 'lab', 'date_created', 'uuid']
+    else:
+        ignored_qc_fields = []
+    # for each file
+    for associated_file in associated_files:
+        if associated_file in data:
+            for entry in data[associated_file]:
+                if entry.get('quality_metric'):
+                    # check if it is a list of qc metrics
+                    if entry['quality_metric']['display_title'].startswith('QualityMetricQclist'):
+                        qc_metric_list = get_metadata(entry['quality_metric']['uuid'], key=key, ff_env=ff_env)
+                        if not qc_metric_list.get('qc_list'):
+                            continue
+                        for qc in qc_metric_list['qc_list']:
+                            qc_uuid = qc['value']['uuid']
+                            qc_info = {qc_uuid: {}}
+                            qc_meta = get_metadata(qc_uuid, key=key, ff_env=ff_env)
+                            qc_info[qc_uuid]['values'] = {k: v for k, v in qc_meta.items() if k not in ignored_qc_fields}
+                            qc_info[qc_uuid]['source_file_association'] = associated_file if associated_file != 'files' else 'raw_file'
+                            qc_info[qc_uuid]['source_file'] = entry['accession']
+                            qc_info[qc_uuid]['source_file_type'] = entry['file_type_detailed']
```

Comment:
It's a style issue, so not a big deal, but did you experiment with writing this in a way that is more visually clear, like this (untested):
```
qc_meta = get_metadata(qc_uuid, key=key, ff_env=ff_env)
qc_info = {
    qc_uuid: {
        'values': {k: v for k, v in qc_meta.items() if k not in ignored_qc_fields},
        'source_file_association': associated_file if associated_file != 'files' else 'raw_file',
        'source_file': entry['accession'],
        'soure_file_type': entry['file_type_detailed']
    }
}
```
The disadvantage is that if you have an error, it's a little harder to debug. Your way has a lot of friendly variables that can be seen separately when stepping. But the result is not as clear. A possible way to combine the two styles is to first compute these as local variables and then put it together:
```
values = ...
source_file_association = ...
...
qc_info = {
    qc_uuid: {
        'values': values,
       ...etc.
    }
}
```
I just think sometimes one sees big long columns of assignments like this and gets no easy sense of what's being done with it all. But I admit that's subjective.